src/web100srv.c

```diff
@@ -1951,375 +1837,335 @@ void NDT_server_main_loop(SSL_CTX *ssl_context, int listenfd) {
+      // queuing of multiple incoming clients
+
+      log_println(5, "Parent process spawned child = %d", chld_pid);
+      log_println(5, "Parent thinks pipe() returned fd0=%d, fd1=%d",
+                  chld_pipe[0], chld_pipe[1]);
+
+      // Close the open resources that should only be used by the child.
+      close(chld_pipe[0]);
+      close(ctlsockfd);
+
+      // Check to see if we have more than max_clients waiting in the queue. If
+      // so, tell them to go away.
+      if (((multiple == 0) && (waiting >= (max_clients - 1))) ||
+          ((multiple == 1) && (waiting >= ((4 * max_clients) - 1)))) {
+        log_println(0,
+                    "Too many clients/mclients (%d) waiting to be served, "
+                    "Please try again later.",
+                    chld_pid);
+        child_signal = SRV_QUEUE_SERVER_BUSY;
+        write(child_pipe[1], &child_signal, sizeof(int));
+        // Whether the write succeeds or not, we are done with this child. It
+        // will either kill itself in response to this message, or it will kill
+        // itself with its own watchdog timer. Either way, not our problem
+        // anymore. Send the message, close the pipe, and move on.
+        close(child_pipe[1]);
+        if (new_child != NULL) {
+          log_println(6, "Too many clients, freeing child=0x%x", new_child);
+          free(new_child);
         }
-        // socket interrupt, retry
+        continue;
+      }
+      // socket interrupt, retry
+      while ((retcode = sem_wait(&ndtq)) == -1 && errno == EINTR) {
+        log_println(6, "Waiting for ndtq semaphore to free - 1");
+        continue;
+      }
+      log_println(6, "creating new child - semaphore locked");
+      new_child->pid = chld_pid;
+      strlcpy(new_child->addr, rmt_addr, sizeof(new_child->addr));
+      strlcpy(new_child->host, rmt_host, sizeof(new_child->host));
+
+      // compute start time based in the number of waiting clients
+      // set other properties on this child process
+      if (multiple == 0) {
+        new_child->stime = tt + (waiting * 45);
+      } else {
+        new_child->stime = tt + ((waiting / max_clients) * 45);
+      }
+      new_child->qtime = tt;
+      new_child->pipe = chld_pipe[1];
+      new_child->running = 0;
+      new_child->next = NULL;
+      sem_post(&ndtq);
+      log_println(6, "Free'd ndtq semaphore lock - 1");
+      if (multiple == 1) mwaiting++;
+      mchild = new_child;
+
+      if ((testing == 1) && (queue == 0)) {
+        log_println(3,
+                    "queuing disabled and testing in progress, tell client no");
+        child_signal = SRV_QUEUE_SERVER_BUSY;
+        write(child_pipe[1], &child_signal, sizeof(int));
+        close(chld_pipe[1]);
+        log_println(6, "no queuing, free new_child=0x%x", new_child);
+        free(new_child);
+        continue;
+      }
+
+      waiting++;  // new client has arrived and is queue-able.
+                  // increment waiting
```

Comment:
Seems like in other places we're moving these kinds of comments to the preceding line, so suggest same here.